components/builder-depot/src/server.rs

```diff
@@ -1241,6 +1241,73 @@ fn list_package_versions(req: &mut Request) -> IronResult<Response> {
     }
 }
 
+fn list_origin_packages_for_version(req: &mut Request) -> IronResult<Response> {
+
+    let (start, stop) = match extract_pagination(req) {
+        Ok(range) => range,
+        Err(response) => return Ok(response),
+    };
+
+    let ident = {
+        let params = req.extensions.get::<Router>().unwrap();
+        ident_from_params(params).to_string()
+    };
+
+    let packages: RouteResult<OriginPackagesForVersionResponse>;
+    let mut request = OriginPackagesForVersionRequest::new();
+    request.set_start(start as u64);
+    request.set_stop(stop as u64);
+
+    request.set_ident(OriginPackageIdent::from_str(ident.as_str()).expect(
+        "invalid package identifier",
+    ));
+
+    packages = route_message::<OriginPackagesForVersionRequest, OriginPackagesForVersionResponse>(req, &request);
+
+    match packages {
+        Ok(packages) => {
+            debug!(
+                "list_origin_packages_for_version start: {}, stop: {}, total count: {}",
+                packages.get_start(),
+                packages.get_stop(),
+                packages.get_count()
+            );
+
+            let body = package_results_json(
+                &packages.get_packages().to_vec(),
+                packages.get_count() as isize,
+                packages.get_start() as isize,
+                packages.get_stop() as isize,
+            );
+
+            let mut response =
+                if packages.get_count() as isize > (packages.get_stop() as isize + 1) {
+                    Response::with((status::PartialContent, body))
```

Comment:
Yeah, I don't think 206 is the right status for these, either, particularly since we don't seem to traffic in `Content-Range` headers anymore (it looks like we [used to](https://github.com/habitat-sh/habitat/pull/1387).) I'll open an issue to change them to 200s.